machineconfiguration/v1/types.go

```diff
@@ -503,8 +503,73 @@ type MachineConfigPoolStatus struct {
 	// +listType=atomic
 	// +optional
 	CertExpirys []CertExpiry `json:"certExpirys"`
+
+	// poolSynchronizersStatus is the status of the machines managed by the pool synchronizers.
+	// +openshift:enable:FeatureGate=PinnedImages
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	PoolSynchronizersStatus []PoolSynchronizerStatus `json:"poolSynchronizersStatus"`
+}
+
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.updatedMachineCount", message="machineCount must be greater than or equal to updatedMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.availableMachineCount", message="machineCount must be greater than or equal to availableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.unavailableMachineCount", message="machineCount must be greater than or equal to unavailableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.readyMachineCount", message="machineCount must be greater than or equal to readyMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.availableMachineCount >= self.readyMachineCount", message="availableMachineCount must be greater than or equal to readyMachineCount"
+type PoolSynchronizerStatus struct {
+	// name is the name of a pool synchronizer. Must adhere to
+	// RFC-1123 (https://tools.ietf.org/html/rfc1123).
+	// Made up of one of more period-separated (.) segments, where each segment
+	// consists of alphanumeric characters and hyphens (-), must begin and end
+	// with an alphanumeric character, and is at most 63 characters in length.
+	// The total length of the name must not exceed 253 characters.
+	// +openshift:enable:FeatureGate=PinnedImages
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// poolSynchronizerType describes the type of the pool synchronizer.
+	// +kubebuilder:validation:Required
+	PoolSynchronizerType PoolSynchronizerType `json:"poolSynchronizerType"`
```

Comment:
For a PinnedImageSet give me an example of the name and poolSynchronizerType please.  If they're the same, why have both?

Reply:
I see a possibility of using the name as a ref and the type as explicit relationship. Or another usecase is maybe it makes sense to group the sets into a group which has a name.

name: worker-set <-report on single pinnedImageSet
type: PinnedImageSet

name: rendered-worker-set <-report on a set that is a combination of more than one set
type: PinnedImageSet

Reply:
without a name we would be bound to a single synchronizer per type which may or may not be desired depending on the type or future type.

Reply:
so the mapkey is actually "name,poolSynchronizerType" in the status field?

Reply:
correct it should be

Reply:
After talking with MCO team we dropped the name field because the feeling is that a single synchronizer per type is reasonable.